components/brave_wallet/common/fil_address.cc

```diff
@@ -0,0 +1,196 @@
+  if ((input - '0') == static_cast<int>(mojom::FilecoinAddressProtocol::BLS))
+    return mojom::FilecoinAddressProtocol::BLS;
+  if ((input - '0') ==
+      static_cast<int>(mojom::FilecoinAddressProtocol::SECP256K1))
+    return mojom::FilecoinAddressProtocol::SECP256K1;
+  return absl::nullopt;
+}
+
+}  // namespace
+
+FilAddress::FilAddress(const std::vector<uint8_t>& bytes,
+                       mojom::FilecoinAddressProtocol protocol,
+                       const std::string& network)
+    : protocol_(protocol), network_(network), bytes_(bytes) {
+  DCHECK(IsValidNetwork(network));
+}
+FilAddress::FilAddress() = default;
+FilAddress::FilAddress(const FilAddress& other) = default;
+FilAddress::~FilAddress() = default;
+
+bool FilAddress::IsEqual(const FilAddress& other) const {
+  return bytes_.size() == other.bytes_.size() &&
+         std::equal(bytes_.begin(), bytes_.end(), other.bytes_.begin()) &&
+         protocol_ == other.protocol_ && network_ == other.network_;
+}
+
+bool FilAddress::operator==(const FilAddress& other) const {
+  return IsEqual(other);
+}
+
+bool FilAddress::operator!=(const FilAddress& other) const {
+  return !IsEqual(other);
+}
+
+// Decodes Filecoin BLS/SECP256K addresses within rules
+// https://spec.filecoin.io/appendix/address/#section-appendix.address.string
+// |------------|----------|---------|----------|
+// |  network   | protocol | payload | checksum |
+// |------------|----------|---------|----------|
+// | 'f' or 't' |  1 byte  | n bytes | 4 bytes  |
+// static
+FilAddress FilAddress::FromAddress(const std::string& address) {
+  if (address.size() != kAddressSizeBLS &&
+      address.size() != kAddressSizeSecp256K)
+    return FilAddress();
+
+  auto protocol = ToProtocol(address[1]);
+  if (!protocol)
+    return FilAddress();
+
+  std::string network{address[0]};
+  if (!IsValidNetwork(network))
+    return FilAddress();
+
+  std::string address_hash_decoded{
+      base32::Base32Decode(base::ToUpperASCII(address.substr(2)))};
+  if (address_hash_decoded.empty())
+    return FilAddress();
+
+  std::string payload_piece{address_hash_decoded.substr(
+      0, address_hash_decoded.size() - kChecksumSize)};
+  std::vector<uint8_t> payload(payload_piece.begin(), payload_piece.end());
+  return FilAddress::FromPayload(payload, protocol.value(), network);
+}
+
+// Creates FilAddress from SECP256K uncompressed public key
+// with specified protocol and network
+// https://spec.filecoin.io/appendix/address/#section-appendix.address.string
+// static
+FilAddress FilAddress::FromUncompressedPublicKey(
+    const std::vector<uint8_t>& uncompressed_public_key,
+    mojom::FilecoinAddressProtocol protocol,
+    const std::string& network) {
+  if (protocol != mojom::FilecoinAddressProtocol::SECP256K1)
+    return FilAddress();
+  if (uncompressed_public_key.empty())
+    return FilAddress();
+  auto payload = BlakeHash(uncompressed_public_key, kPublicKeySizeSecp256K);
```

Comment:
kPublicKeySizeSecp256K is not the size of secp256k1 public key, this is the hash length, please use a proper name here.

Reply:
done